packages/pages/src/dev/server/middleware/serverRenderRoute.ts

```diff
@@ -57,15 +57,19 @@ export const serverRenderRoute =
           featuresConfig,
           dynamicGenerateData,
           projectStructure,
-        }
-      );
+        });
+
+      if (render) {
+        res.status(200).set({ "Content-Type": "text/html" }).end(render(props));
```

Comment:
the content type isn't necessarily text/html.

In the issue, they were trying to render robots.txt